### ISSUE TITLE
Not able to compile the sketch file.

### ISSUE THREAD

**Issue by codezuma** [MrYsLab/Telemetrix4Esp32#6]
Version -  2.1.0.

## While compiling the wifi code TELEMETRIX4ESP32WIFI.ino 
- I have tried both the versions commenting and without commenting the LED_BUILTIN_SUPPORTED,DAC_SUPPORTED

## Steps to Re create 
1. Installed the telemetrix4esp32
2. files -> examples -> telemtrix -> TELEMETRIX4ESP32WIFI
3. changed the ssid and password to my local network (saved the file)
4. commented the  LED_BUILTIN_SUPPORTED,DAC_SUPPORTED (optional)

I am getting this error while compiling
![image](https://github.com/user-attachments/assets/96242766-bbe5-4e04-af87-649c7275b17c)

```C:\Users\OMEN\Documents\Arduino\Telemetrix4Esp32WIFI\Telemetrix4Esp32WIFI.ino: In function 'void set_pin_mode()':
C:\Users\OMEN\Documents\Arduino\Telemetrix4Esp32WIFI\Telemetrix4Esp32WIFI.ino:613:7: error: 'ledcSetup' was not declared in this scope
  613 |       ledcSetup(channel, frequency, resolution);
      |       ^~~~~~~~~
C:\Users\OMEN\Documents\Arduino\Telemetrix4Esp32WIFI\Telemetrix4Esp32WIFI.ino:614:7: error: 'ledcAttachPin' was not declared in this scope; did you mean 'ledcAttach'?
  614 |       ledcAttachPin(pin, channel);
      |       ^~~~~~~~~~~~~
      |       ledcAttach
C:\Users\OMEN\Documents\Arduino\Telemetrix4Esp32WIFI\Telemetrix4Esp32WIFI.ino: In function 'void analog_out_attach()':
C:\Users\OMEN\Documents\Arduino\Telemetrix4Esp32WIFI\Telemetrix4Esp32WIFI.ino:625:3: error: 'ledcAttachPin' was not declared in this scope; did you mean 'ledcAttach'?
  625 |   ledcAttachPin(command_buffer[0], command_buffer[1]);
      |   ^~~~~~~~~~~~~
      |   ledcAttach
C:\Users\OMEN\Documents\Arduino\Telemetrix4Esp32WIFI\Telemetrix4Esp32WIFI.ino: In function 'void analog_out_detach()':
C:\Users\OMEN\Documents\Arduino\Telemetrix4Esp32WIFI\Telemetrix4Esp32WIFI.ino:630:3: error: 'ledcDetachPin' was not declared in this scope; did you mean 'ledcDetach'?
  630 |   ledcDetachPin(command_buffer[0]);
      |   ^~~~~~~~~~~~~
      |   ledcDetach

exit status 1

Compilation error: 'ledcSetup' was not declared in this scope```

- Additional information I have tried running the

**Comment by codezuma** [comment]
@MrYsLab .

**Comment by MrYsLab** [comment]
The ESP32 folks changed their API. This is documented in [this issue report.](https://github.com/espressif/arduino-esp32/issues/10309#issuecomment-2336961421)

Temporarily, you may wish to fall back to version 2.0.17, as mentioned in the issue.

I will need a little time to read the migration guide and see if I can make the library compatible with version 3.

Microsoft (which owns GitHub) made changes to GitHub, and I did not receive a notification of the new release. I had to select a new "watch" option, and hopefully, I will receive notifications in the future.

Thanks for letting me know.

**Comment by codezuma** [comment]
@MrYsLab  I am able to compile code after downgrading the version to 2.0.17. as you mentioned.

After running the code and scratch editor is not able to run code on esp32.
 
I have added an ip block tried connecting to ip I got from the Ardino IDE

![image](https://github.com/user-attachments/assets/e5783582-7405-4d9b-b0b5-3225ce961671)

started the s32 library websockets are running fine and scratch editor is connected to it. 

![image](https://github.com/user-attachments/assets/4189602e-dc0f-4d41-911e-ba744f5d611f)

I cannot write that digital pin 13.

- Additional Info

I writing the pin with Ardino IDE directly it seems to be working fine.

**Comment by codezuma** [comment]
Update : seems to be working fine after I changed the wifi network

**Comment by MrYsLab** [comment]
I have modified both the WIFI and BLE libraries so that they now compile with version 3.0.7 of the ESP boards library for Arduino. 

I must remove the attach_pin_to_analog_channel method from the Python API since the ESPressif library no longer supports it. So, there will be a bump in release for [telemetrix-esp32](telemetrix-esp32).

The Scratch extension will not need changes since it does not use the attach_pin_to_analog_channel, but I will need to test this to ensure it is true.

After I test the changes, I will release the next version. This may take a few days.

**Comment by MrYsLab** [comment]
I updated and published version 3.0.0 of this library and version 2.0.0 of[ telemtrix-esp32.](https://github.com/MrYsLab/telemetrix-esp32)

All should compile and work if both of the libraries mentioned above are updated.
